### PR TITLE
Update Tradheli connections page to include Motor Interlock

### PR DESCRIPTION
**Pull request by bnsgeyer**
This PR updates the wiki that discusses connecting the controller to the receiver and servos.  Made some pretty big changes to bring the page up to date to include a discussion on motor interlock.

### CODE REVIEW

File: copter/source/docs/traditional-helicopter-connecting-apm.rst [hunk 1]
```diff
@@ -10,63 +10,54 @@ Autopilot Info
...
-+--------------+-------------+
-| 5            | Flight Modes|
-+--------------+-------------+
-| 6            | Tuning      |
-+--------------+-------------+
-| 7            | Aux         |
-+--------------+-------------+
-| 8            | Throttle    |
-+--------------+-------------+
+The RC input for many ardupilot compatible autopilots is either PPM SUM (8 channels) or S.Bus (up to 18 channels).  Some controllers also accept Spektrum satellite receivers.  For receivers that only output PWM, a PPM encoder is required to connect to the autopilot.
```
Comment (Hwurzburg) [inline copter/source/docs/traditional-helicopter-connecting-apm.rst]: might be nice to state that the minimum # RC inputs channels is (x....think its 6?) and outputs is (y...thinks its 5?) since we have many different kinds of autopilots and rc systems

File: copter/source/docs/traditional-helicopter-connecting-apm.rst [hunk 2]
```diff
@@ -10,63 +10,54 @@ Autopilot Info
...
 
 The output pins on most controllers for SERVO's 1 thru 8 are labled Main Out:
 
 .. image:: ../images/PH21_2.jpg
     :target: ../_images/PH21_2.jpg
 
-Motor 1, which is normally the left front servo on your helicopter goes to pin 1.
-Motor 2, which is normally the right front servo goes to pin 2.  Motor 3, which 
-is normally the rear (elevator) servo goes to pin 3. Tail servo goes to pin 4.
+The :ref:`autopilot output functions wiki <common-rcoutput-mapping>` shows the complete list of servo output functions.  The default swashplate is H3-120 where Motor 1, left front servo, goes to pin 1; Motor 2, right front servo goes to pin 2; and Motor 3, rear (elevator) servo, goes to pin 3. See the :ref:`swashplate setup wiki <traditional-helicopter-swashplate-setup>` for more details.
```
Comment (Hwurzburg) [inline copter/source/docs/traditional-helicopter-connecting-apm.rst]: I think we should change all references from "pins" to "outputs"....might be good to re-iterate that Motor 3 corresponds to having SERVOn_FUNCTION = 35 where n is the output number.  Different autopilots have different marking for their outputs and some dont have "pins" marked at all...

File: copter/source/docs/traditional-helicopter-connecting-apm.rst (hunk 1 above)
Comment (ChristopherOlson) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: Minimum input channels is 8, min output is 6

File: copter/source/docs/traditional-helicopter-connecting-apm.rst (hunk 2 above)
Comment (ChristopherOlson) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: Yes, this is true. The outputs are SERVO. The inputs for aux ports are "pins".

File: copter/source/docs/traditional-helicopter-connecting-apm.rst [hunk 3]
```diff
@@ -78,22 +69,17 @@ RC Calibration
...
-It is recommended to calibrate the compasses at this time as well. This is the
-same as all other vehicles.
+It is recommended to calibrate the compasses at this time as well. This is the same as all other vehicles.
 :ref:`See this topic <common-compass-calibration-in-mission-planner>`.
 
 Accelerometer Calibration
 -------------------------
-If the accelerometers were not calibrated on the bench prior to installation it
-must be calibrated before proceeding.
+If the accelerometers were not calibrated on the bench prior to installation it must be calibrated before proceeding.
```
Comment (ChristopherOlson) [inline copter/source/docs/traditional-helicopter-connecting-apm.rst]: It might be worth a note here that accelerometer calibration in helicopters is very difficult with the controller mounted in the frame. With large 800 class machines that weigh 30 lbs or more, it almost impossible. Highly recommended to perform accel calibration before installing the flight control. It can be done with the control connected to the GCS with USB cable before installation.

File: copter/source/docs/traditional-helicopter-connecting-apm.rst [hunk 4]
```diff
@@ -10,63 +10,54 @@ Autopilot Info
...
...se Motor Interlock.  This feature adds an extra layer of safety when working with helicopters.  Motor interlock enables the motor to drive the rotor/tailrotor.  This is similar to throttle hold in RC helicopters.  Motor interlock enabled (throttle hold off) means the motor is allowed to drive the rotor/tailrotor and the rotor speed control handles the rotor runup/shutdown.  Motor interlock disabled (throttle hold on) means the motor is not allowed to drive the rotor/tailrotor.  In order to arm the helicopter, the motor interlock must be disabled (throttle hold on). In ArduCopter 3.6 and earlier, the motor interlock and RC passthrough mode is tied to RC channel 8 only.  The RC transmitter channel 8 must have the PWM within 10 pwm of the RC8_MIN for motor interlock disabled (throttle hold on).  All other PWM values will set motor interlock enabled (throttle hold off).  In ArduCopter 4.0, the RCn_Option parameter can be set to motor interlock for a user selectable channel.  The transmitter channel on which the motor interlock is set requires the PWM to be low (<1200 PWM) for motor interlock disabled (throttle hold on) and above 1200 PWM for motor interlock enabled (throttle hold off).
```
Comment (ChristopherOlson) [inline copter/source/docs/traditional-helicopter-connecting-apm.rst]: I have tested this, but is there any limitation on using, say Channel 5 or 6, for throttle control in 4.0 when those channels are normally pre-set for flight modes and tuning? I know those can be re-assigned as well. But what setup problems occur if the throttle is assigned to 5, when 5 is also pre-assigned to flight modes? Does the system spit out an error message for that?

Comment (Hwurzburg) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: easy to test in SITL I think, assuming there is a trad heli sitl model...
Chris, can I call you or skype you to discuss some things?

File: copter/source/docs/traditional-helicopter-connecting-apm.rst (hunk 1 above)
Comment (Hwurzburg) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: you mean you cant fly without tuning or aux? I know setup is going to be difficult without them, but you cant fly without them?
5 outputs are: 3 for swash/collective, rsc, yaw....what is the 6th? you have to have yaw mode/tuning? bear in mind I am coming from flybar days on helis...
just coming from the simplest point of view...

Comment (ChristopherOlson) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: Technically, in 4.0 I think a 6 channel RC radio will work. The RSC, or throttle, used to be hard-wired to channel 8 in heli.cpp. That has changed in 4.0.

I guess if a controller exists with only 5 outputs, it could probably be used. Is there any like that? Even the Pixracer has 6.

Comment (Hwurzburg) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: with the new board alt configs there could be...I made a custom omnibusf4pro compile for a guy that changed one of the 6 pwms to get another uart output, that might be a standard alternate in the future...no telling what a future board might happen...but the real point is understanding what is or is not possible...I have an autonomous rover that only uses two channels on rc...I have to change modes with my phone gcs...just trying to understand what is or is not possible..

File: copter/source/docs/traditional-helicopter-connecting-apm.rst (hunk 3 above)
Comment (Hwurzburg) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: I even do that on my small planes and multicopters....its always easier to cal first time on the bench....with Tridge's new align on a landmark and enter the heading via gcs one button cal for compass, doing the compass cal with a 30lb heli will get a lot easier also...

File: copter/source/docs/traditional-helicopter-connecting-apm.rst [hunk 5]
```diff
@@ -17,7 +17,7 @@ Before you begin connecting the system it is recommended to review the docs for
 Overview of servo, and RX connection
 ====================================
 
-The RC input for many ardupilot compatible autopilots is either PPM SUM (8 channels) or S.Bus (up to 18 channels).  Some controllers also accept Spektrum satellite receivers.  For receivers that only output PWM, a PPM encoder is required to connect to the autopilot.  For all helicopter frames,the RC radio requires at least 6 channels.  Traditional helicopter and quadheli frames require the flight controller have at least 5 channels. The dual helicopter frame requires the flight controller have at least 7 channels. 
+The RC input for many ardupilot compatible autopilots is either PPM SUM (8 channels) or S.Bus (up to 18 channels).  Some controllers also accept Spektrum satellite receivers.  For receivers that only output PWM, a PPM encoder is required to connect to the autopilot however this is not recommended.  For all helicopter frames, an RC radio that has at least 6 channels is required.  For flight controllers, traditional helicopter and quadheli frames require one that has at least 5 output channels. The dual helicopter frame requires the flight controller have at least 7 output channels.
```
Comment (Hwurzburg) [inline copter/source/docs/traditional-helicopter-connecting-apm.rst]: Bill, why did you add this? as far as I know, using a PWM RC RX with an encoder is a valid connection method.....other than that I think its ready to merge...

Comment (bnsgeyer) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: @Hwurzburg the PPM encoder is a valid connection but I have seen times where it doesn’t keep up and you see the aircraft not respond to your input. I don’t have solid proof but I don’t recommend them as a a connection method. I know for Futaba, at least, you can buy a really inexpensive receiver ($70USD) that has SBus output that can be used instead of the PPM encoder.

Comment (Hwurzburg) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: yes, Sbus RX for Frsky are $11....but I know of no technical reason why a PPM encoder should be dis-recommended....its maximum delay of one frame (20ms) is unnoticeable..early days of PPM had a few units out there that did not handle all frame widths correctly....but that is a case of defective hardware design early on...we currently  recommend the encoder for other vehicles for those poor souls with old equipment....if you have other than anecodotal evidence, then we should investigate and dis-recommend for all vehicles...

Comment (MidwestAire) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: Henry, PPM encoders are ok for multicopters. But not for helicopters, especially in acro. Pilots demand more crisp control response for helicopters, which PPM encoders don't deliver. We have never recommended them for heli's.

Comment (MidwestAire) [reply copter/source/docs/traditional-helicopter-connecting-apm.rst]: > but I know of no technical reason why a PPM encoder should be dis-recommended....its maximum delay of one frame (20ms) is unnoticeable.

I is very noticeable with helicopters in acro. You combine the latency of an older pwm output receiver with a PPM encoder and the helicopter handles like it's "loose" with all the linkages and swashplate bearing worn out. They are fine for all other vehicle types. It is just the nature of helicopters. Even servos used on airplanes cannot be used on helicopter cyclic.

**Comment by Hwurzburg** [comment]
Merging
